scene/resources/particle_process_material.cpp

```diff
@@ -986,12 +982,14 @@ void ParticleProcessMaterial::_update_shader() {
 		code += "		vec3 noise_direction = get_noise_direction(TRANSFORM[3].xyz);\n";
 		// The following snippet causes massive performance hit. We don't need it as long as collision is disabled.
 		// Refer to GH-83744 for more info.
```

Comment:
I don't know if the change in this PR is correct, but if it is this comment might warrant being further tweaked. It used to sound like the whole `if` block was expensive, but now it seems reduced to only checking for `COLLIDED` is expensive. That should be made explicit.

Reply:
Agreed. And there is a minor tweak I miss, I will update it soon.